### PR TITLE
Explicit casting of vector sizes

### LINKED ISSUE #927: GPU inline helpers
The GPU backends have a few static inline helper functions that return values instead of error codes but use CeedChkBackend internally. CeedChk returns an error code if an error is found, violating the assumed return type for these functions

### PR DESCRIPTION
**Pull request by jeremylt**
I added some explicit casting around `CeedSize` for CeedVector lengths

Closes #927

### CODE REVIEW

File: backends/blocked/ceed-blocked-operator.c [hunk 1]
```diff
@@ -94,23 +95,24 @@ static int CeedOperatorSetupFields_Blocked(CeedQFunction qf,
     switch(eval_mode) {
     case CEED_EVAL_NONE:
       ierr = CeedQFunctionFieldGetSize(qf_fields[i], &size); CeedChkBackend(ierr);
-      ierr = CeedVectorCreate(ceed, Q*size*blk_size, &q_vecs[i]);
-      CeedChkBackend(ierr);
+      q_size = Q*size*blk_size;
```
[the remark this answers is not in the document]
Comment (jeremylt) [reply backends/blocked/ceed-blocked-operator.c]: Thanks, updated

File: backends/blocked/ceed-blocked-operator.c [hunk 2]
```diff
@@ -94,23 +95,24 @@ static int CeedOperatorSetupFields_Blocked(CeedQFunction qf,
     switch(eval_mode) {
     case CEED_EVAL_NONE:
       ierr = CeedQFunctionFieldGetSize(qf_fields[i], &size); CeedChkBackend(ierr);
-      ierr = CeedVectorCreate(ceed, Q*size*blk_size, &q_vecs[i]);
-      CeedChkBackend(ierr);
+      q_size = (CeedSize)Q*size*blk_size;
+      ierr = CeedVectorCreate(ceed, q_size, &q_vecs[i]); CeedChkBackend(ierr);
```
Comment (jedbrown) [inline backends/blocked/ceed-blocked-operator.c:99]: Your choice here. It isn't necessary to make the temp variables, but they don't hurt.

**Review by jedbrown — APPROVED** [review]
This one is a real bug if there's ever an error.
```
/home/jed/src/libCEED/backends/cuda-ref/ceed-cuda-ref-qfunctioncontext.c: In function ‘bytes’:
./include/ceed/backend.h:103:105: warning: unsigned conversion from ‘int’ to ‘size_t’ {aka ‘long unsigned int’} changes value from ‘-2’ to ‘18446744073709551614’ [-Wsign-conversion]
  103 | #define CeedChkBackend(ierr) do { int ierr_ = ierr; if (ierr_) { if (ierr_ > CEED_ERROR_SUCCESS) return CEED_ERROR_BACKEND; else return ierr_; } } while (0)
      |                                                                                                         ^~~~~~~~~~~~~~~~~~
```

The relevant code
```c
static inline size_t bytes(const CeedQFunctionContext ctx) {
  int ierr;
  size_t ctxsize;
  ierr = CeedQFunctionContextGetContextSize(ctx, &ctxsize); CeedChkBackend(ierr);
  return ctxsize;
}
```
then
```c
    ierr = cudaMalloc((void **)&impl->d_data_owned, bytes(ctx));
```

File: backends/blocked/ceed-blocked-operator.c (hunk 2 above)
Comment (jeremylt) [reply backends/blocked/ceed-blocked-operator.c:99]: I added it because it made the casing and the purpose of the value a bit clearer to me, personally.

**Pull request merged**